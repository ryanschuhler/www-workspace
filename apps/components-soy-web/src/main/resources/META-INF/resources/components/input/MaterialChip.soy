{namespace MaterialChip}

/**
 * MaterialChip component
 */
{template .render}
  {@param label: string}
  {@param? deletable: bool}
  {@param? deleteButtonClicked_: any}
  {@param? elementClasses: string}

  {let $classes kind="text"}
    mdl-chip

    {if $deletable}
      {sp}mdl-chip--deletable
    {/if}

    {if $elementClasses}
      {sp}{$elementClasses}
    {/if}
  {/let}

  <span class="{$classes}">
    <span class="mdl-chip__text">{$label}</span>

    {if $deletable}
      <button type="button" class="mdl-chip__action" onClick="{$deleteButtonClicked_}">
        {call Icon.render}
          {param isMaterialIcon: true /}
          {param symbol: 'cancel' /}
        {/call}
      </button>
    {/if}
  </span>
{/template}


