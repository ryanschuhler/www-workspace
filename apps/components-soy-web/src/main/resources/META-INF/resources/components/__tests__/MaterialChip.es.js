'use strict';

import MaterialChip from '../input/MaterialChip.es';

let component;

describe('MaterialChip', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  it('should render', () => {
    component = new MaterialChip({
      label: 'test'
    });

    expect(component).toMatchSnapshot();
  });
});