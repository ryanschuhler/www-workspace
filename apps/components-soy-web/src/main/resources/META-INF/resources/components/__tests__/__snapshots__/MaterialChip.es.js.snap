// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialChip should render 1`] = `
<span class="mdl-chip">
  <span class="mdl-chip__text">test</span>
</span>
`;
